notes/2025-xx-xx-insecurities-remedies-i.md

```diff
@@ -0,0 +1,410 @@
+Here are a few that are worth mentioning while you work on hardening your
+services.
+
+##### InaccessiblePaths
+
+`InaccessiblePaths` is a very useful directive, which I did not about until
+recently, that might come in handy if the service requires access to a lot of
+directives. In which case, you may manually add _sensitive_ directories that you
+want hidden at all costs.
+
+##### DynamicUsers
+
+Systemd _system_ services (as opposed to user services) run as root unless they
+are explicitly given a user to run as. `DynamicUsers` is a special directive
+that can help you with _dynamically_ (look I said the thing!) creating separate
+user accounts for each instance of the service. Each of these unique users runs
+their own instance of the service, providing a high level of isolation between
+different processes. This not only enhances security by limiting the potential
+impact of a compromised process but also improves resource management by
+isolating each instance's file system access. Although the impact of
+`DynamicUsers` on a service's exposure score is low, it is a very handy
+directive that you might consider if root privileges are not necessary for oyur
+service.
+
+##### SystemCallFilter
+
+One noteworthy directive is `SystemCallFilter`. As its name indicates, this
+Directive allows restricting syscalls that a service can call. This is very
+tricky to work with, and you can get out of hand as the process gains new
+features over time. Systemd, to make your life a _bit_ easier, includes
+so-called "groupings"-- several groups of system calls, all prepended with `@`.
+
+`@swap`, `@reboot`, `@memlock` and `@raw-io` are some examples of those groups.
+You may find a more detailed explanation
+[on linux-audit.com](https://linux-audit.com/systemd/settings/units/systemcallfilter/)
+
+##### IP Accounting
+
+With Systemd 235, Systemd was granted ability to track network traffic
+statistics for individual services or units. It allows administrators to monitor
+bandwidth usage, packet counts, and other network-related metrics associated
+with specific systemd services. This feature is implemented through the
+`IPAddressAllow` and `IPAddressDeny` directives in the `[Service]`section of a
+service unit file.
+
+[awesome article on IP accounting]: https://0pointer.net/blog/ip-accounting-and-access-lists-with-systemd.html
+
+By enabling IP accounting, systemd can provide detailed insights into network
+activity, facilitating better network management, troubleshooting, and
+performance optimization for services running under its control. Network
+security is a little out of my scope today, but I encourage you to read the
+[awesome article on IP accounting]
+
+#### Application
+
+From the previous section, you should have a basic understanding of directives
+we will be using to harden services. Now let's take a look at the application of
+those directives.
+
+> Systemd's error messages for when a service is misconfigured can be vague or
```

Comment:
I'd cite this quote as well.

Reply:
Still not a quote, I say a lot of things.